src/main/java/org/candlepin/model/OwnerContentCurator.java

```diff
@@ -428,47 +428,93 @@ public void updateOwnerContentReferences(Owner owner, Map<String, String> conten
+        this.updateOwnerContentEnvironments(owner, contentUuidMap);
+    }
 
-        // FIXME: remove usage of bulkSQLUpdate. While it's a clever use of an SQL CASE to avoid
-        // a ton of individual update statements, it still runs over the parameter limit in some
-        // circumstances and the job could be better done with a temporary table and a couple
-        // join-updates.
+    /**
+     * Part of the updateOwnerContentReferences operation; updates the owner to content join table.
+     * See updateOwnerContentReferences for additional details.
+     *
+     * @param owner
+     * @param contentUuidMap
+     */
+    private void updateOwnerContentJoinTable(Owner owner, Map<String, String> contentUuidMap) {
+        String sql = "UPDATE " + OwnerContent.DB_TABLE + " SET content_uuid = :updated " +
+            "WHERE content_uuid = :current AND owner_id = :owner_id";
 
-        for (Map<String, String> block : this.partitionMap(contentUuidMap)) {
-            Map<String, Object> criteria = new HashMap<>();
-            Map<Object, Object> uuidMap = Map.class.cast(block);
-            criteria.put("content_uuid", block.keySet());
-            criteria.put("owner_id", owner.getId());
+        Query query = this.getEntityManager()
+            .createNativeQuery(sql)
+            .setParameter("owner_id", owner.getId());
 
-            // owner content
-            int count = this.bulkSQLUpdate(OwnerContent.DB_TABLE, "content_uuid", uuidMap, criteria);
-            log.info("{} owner-content relations updated", count);
+        int count = 0;
+        for (Map.Entry<String, String> entry : contentUuidMap.entrySet()) {
+            count += query.setParameter("current", entry.getKey())
+                .setParameter("updated", entry.getValue())
+                .executeUpdate();
+        }
 
-            // environment content
-            String jpql = "SELECT e.id FROM Environment e WHERE e.owner.id = :owner_id";
+        log.debug("{} owner-content relations updated", count);
+    }
 
-            List<String> envIds = this.getEntityManager()
-                .createQuery(jpql, String.class)
-                .setParameter("owner_id", owner.getId())
+    /**
+     * Part of the updateOwnerContentReferences operation; updates the owner to content join table.
+     * See updateOwnerContentReferences for additional details.
+     *
+     * @param owner
+     * @param contentUuidMap
+     */
+    private void updateOwnerContentEnvironments(Owner owner, Map<String, String> contentUuidMap) {
+        EntityManager entityManager = this.getEntityManager();
+
+        String jpql = "SELECT env.id, content.uuid FROM EnvironmentContent ec " +
+            "JOIN ec.content content " +
+            "JOIN ec.environment env " +
+            "WHERE env.owner.id = :owner_id " +
+            "  AND content.uuid IN (:content_uuids)";
+
+        Query query = entityManager.createQuery(jpql)
+            .setParameter("owner_id", owner.getId());
```

Comment:
Possible NPE here, we may want to check that owner isn't null first. It looks like this may also need to be done for the methods: 
updateOwnerProductJoinTable
updateOwnerProductPools
updateOwnerProductActivationKeys